src/storage/rdb.cc

```diff
@@ -449,3 +510,188 @@ Status RDB::Restore(const std::string &key, uint64_t ttl_ms) {
   }
   return db_status.ok() ? Status::OK() : Status{Status::RedisExecErr, db_status.ToString()};
 }
+
+StatusOr<uint32_t> RDB::loadTime() {
+  uint32_t t32 = 0;
+  GET_OR_RET(stream_->Read(reinterpret_cast<char *>(&t32), 4));
+  return t32;
+}
+
+StatusOr<uint64_t> RDB::loadMillisecondTime(int rdb_version) {
+  uint64_t t64 = 0;
+  GET_OR_RET(stream_->Read(reinterpret_cast<char *>(&t64), 8));
+  /* before Redis 5 (RDB version 9), the function
+   * failed to convert data to/from little endian, so RDB files with keys having
+   * expires could not be shared between big endian and little endian systems
+   * (because the expire time will be totally wrong). comment from src/rdb.c: rdbLoadMillisecondTime*/
+  if (rdb_version >= 9) {
+    memrev64ifbe(&t64);
+  }
+  return t64;
+}
+
+bool RDB::isEmptyRedisObject(const RedisObjValue &value) {
+  if (auto vec_str_ptr = std::get_if<std::vector<std::string>>(&value)) {
+    return vec_str_ptr->size() == 0;
+  }
+  if (auto vec_mem_ptr = std::get_if<std::vector<MemberScore>>(&value)) {
+    return vec_mem_ptr->size() == 0;
+  }
+  if (auto map_ptr = std::get_if<std::map<std::string, std::string>>(&value)) {
+    return map_ptr->size() == 0;
+  }
+
+  return false;
+}
+
+// Load RDB file: copy from redis/src/rdb.c:branch 7.0, 76b9c13d.
+Status RDB::LoadRdb(uint32_t db_index, bool is_nx) {
+  char buf[1024] = {0};
+  GET_OR_RETWITHLOG(stream_->Read(buf, 9));
+  buf[9] = '\0';
+
+  if (memcmp(buf, "REDIS", 5) != 0) {
+    LOG(WARNING) << "Wrong signature trying to load DB from file";
+    return {Status::NotOK};
```

Comment:
Can we bring the error message into Status?

Reply:
okay